src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -893,11 +893,28 @@ GenerateAllShardStatisticsQueryForNode(WorkerNode *workerNode, List *citusTableI
 	Oid relationId = InvalidOid;
 	foreach_oid(relationId, citusTableIds)
 	{
-		List *shardIntervalsOnNode = ShardIntervalsOnWorkerGroup(workerNode, relationId);
-		char *shardStatisticsQuery =
-			GenerateShardStatisticsQueryForShardList(shardIntervalsOnNode,
-													 useShardMinMaxQuery);
-		appendStringInfoString(allShardStatisticsQuery, shardStatisticsQuery);
+		/*
+		 * Ensure the table still exists by trying to acquire a lock on it
+		 * If function returns NULL, it means the table has been concurrently dropped
+		 * hence we should skip
+		 */
+		Relation relation = try_relation_open(relationId, AccessShareLock);
+		if (relation != NULL)
+		{
+			List *shardIntervalsOnNode = ShardIntervalsOnWorkerGroup(workerNode,
+																	 relationId);
+			char *shardStatisticsQuery =
+				GenerateShardStatisticsQueryForShardList(shardIntervalsOnNode,
+														 useShardMinMaxQuery);
+			appendStringInfoString(allShardStatisticsQuery, shardStatisticsQuery);
+			relation_close(relation, AccessShareLock);
+		}
+		else
+		{
+			ereport(NOTICE, (errmsg(
+								 "relation with OID %u was concurrently dropped, skipping",
```

Comment:
I am thinking that we can remove this NOTICE here while keeping the other one. Because this will be triggered with only citus_shards and I feel like showing this notice message isn't that useful in that case.

Reply:
That makes sense. After all `citus_shards` shows an output and the user knows which tables exist. That also fixes the problem of showing the same warning multiple times that I mentioned [here](https://github.com/citusdata/citus/pull/5155#issuecomment-892113662)

> There is one thing I don't quite like though: we generate shard statistics queries for each worker node, so we will get the notice of relation being dropped concurrently multiple times (the number of worker nodes). However, I believe this is not expected to happen often.

Reply:
Yeah this came to my mind while thinking about the problem you mentioned :)